charts/surrealdb/templates/deployment.yaml

```diff
@@ -57,7 +57,9 @@ spec:
               value: {{ . }}
             {{- end }}
             - name: SURREAL_AUTH
-              value: "{{ .Values.surrealdb.auth }}"
+              value: "{{ .Values.surrealdb.auth | default (not .Values.surrealdb.unauthenticated) }}"
```

Comment:
This change introduces a regression where you can't disable auth on v1 because of this:

```
helm template charts/surrealdb --set surrealdb.unauthenticated= --set surrealdb.auth=false
```

Results in:

```yaml
            - name: SURREAL_AUTH
              value: "true" # SHOULDNT THIS BE FALSE?!
            - name: SURREAL_UNAUTHENTICATED
              value: ""
```

-which is probably different from what we want.

You've added this fallback to SURREAL_AUTH envvar for backward compatibility, which enables you to deploy v1 without changing values, right?

It looks odd to me, but that's how Helm works, as far as I know. See https://github.com/helm/helm/issues/3308

We probably need to rewrite this to [use `if` and `hasKey`](https://github.com/helm/helm/issues/3308#issuecomment-418963739) instead.